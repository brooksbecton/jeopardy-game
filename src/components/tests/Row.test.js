import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";

import { Row } from "./../Row";
import { QuestionCell } from "./../QuestionCell";

describe("Row Component", () => {
  const mockCategories = {
    "0": [
      {
        value: 200,
        category: "a",
        question: "a1",
        answer: "a1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "a",
        question: "a2",
        answer: "a2a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "a",
        question: "a3",
        answer: "a3a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "a",
        question: "a4",
        answer: "a4a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "a",
        question: "a5",
        answer: "a5a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ],
    "1": [
      {
        value: 200,
        category: "b",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "b",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "b",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "b",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "b",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ],
    "2": [
      {
        value: 200,
        category: "c",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "c",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "c",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "c",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "c",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ],
    "3": [
      {
        value: 200,
        category: "d",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "d",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "d",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "d",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "d",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ],
    "4": [
      {
        value: 200,
        category: "e",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "e",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "e",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "e",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "e",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ],
    "5": [
      {
        value: 200,
        category: "f",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 400,
        category: "f",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 600,
        category: "f",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 800,
        category: "f",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      },
      {
        value: 1000,
        category: "f",
        question: "b1",
        answer: "b1a",
        youtubeLink: "",
        imageLink: "",
        isAnswered: false
      }
    ]
  };

  beforeAll(() => {
    configure({ adapter: new Adapter() });
  });

  it("renders correctly", () => {
    const component = shallow(<Row value={200} categories={mockCategories} />);
    expect(component).toMatchSnapshot();
  });

  it("renders 6 question rows", () => {
    const component = shallow(<Row value={200} categories={mockCategories} />);
    expect(component.find("tr").children()).toHaveLength(6);
  });
});
