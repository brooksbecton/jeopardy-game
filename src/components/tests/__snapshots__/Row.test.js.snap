// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row Component renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Row
    categories={
      Object {
        "0": Array [
          Object {
            "answer": "a1a",
            "category": "a",
            "imageLink": "",
            "isAnswered": false,
            "question": "a1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "a2a",
            "category": "a",
            "imageLink": "",
            "isAnswered": false,
            "question": "a2",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "a3a",
            "category": "a",
            "imageLink": "",
            "isAnswered": false,
            "question": "a3",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "a4a",
            "category": "a",
            "imageLink": "",
            "isAnswered": false,
            "question": "a4",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "a5a",
            "category": "a",
            "imageLink": "",
            "isAnswered": false,
            "question": "a5",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
        "1": Array [
          Object {
            "answer": "b1a",
            "category": "b",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "b",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "b",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "b",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "b",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
        "2": Array [
          Object {
            "answer": "b1a",
            "category": "c",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "c",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "c",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "c",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "c",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
        "3": Array [
          Object {
            "answer": "b1a",
            "category": "d",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "d",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "d",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "d",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "d",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
        "4": Array [
          Object {
            "answer": "b1a",
            "category": "e",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "e",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "e",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "e",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "e",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
        "5": Array [
          Object {
            "answer": "b1a",
            "category": "f",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 200,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "f",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 400,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "f",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 600,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "f",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 800,
            "youtubeLink": "",
          },
          Object {
            "answer": "b1a",
            "category": "f",
            "imageLink": "",
            "isAnswered": false,
            "question": "b1",
            "value": 1000,
            "youtubeLink": "",
          },
        ],
      }
    }
    value={200}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          category="0"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
        <Unknown
          category="1"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
        <Unknown
          category="2"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
        <Unknown
          category="3"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
        <Unknown
          category="4"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
        <Unknown
          category="5"
          isAnswered={false}
          openQuestion={undefined}
          value={200}
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "function",
        "props": Object {
          "category": "0",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "function",
        "props": Object {
          "category": "1",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "2",
        "nodeType": "function",
        "props": Object {
          "category": "2",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "3",
        "nodeType": "function",
        "props": Object {
          "category": "3",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "4",
        "nodeType": "function",
        "props": Object {
          "category": "4",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": "5",
        "nodeType": "function",
        "props": Object {
          "category": "5",
          "isAnswered": false,
          "openQuestion": undefined,
          "value": 200,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "tr",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            category="0"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
          <Unknown
            category="1"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
          <Unknown
            category="2"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
          <Unknown
            category="3"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
          <Unknown
            category="4"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
          <Unknown
            category="5"
            isAnswered={false}
            openQuestion={undefined}
            value={200}
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "function",
          "props": Object {
            "category": "0",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "function",
          "props": Object {
            "category": "1",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "2",
          "nodeType": "function",
          "props": Object {
            "category": "2",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "3",
          "nodeType": "function",
          "props": Object {
            "category": "3",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "4",
          "nodeType": "function",
          "props": Object {
            "category": "4",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": "5",
          "nodeType": "function",
          "props": Object {
            "category": "5",
            "isAnswered": false,
            "openQuestion": undefined,
            "value": 200,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "tr",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
